stack/CO_PDO.c

```diff
@@ -992,6 +997,13 @@ void CO_TPDO_process(
                     TPDO->eventTimer = ((uint32_t) TPDO->TPDOCommPar->eventTimer) * 1000;
                 }
             }
+            if(timerNext_us != NULL){
+                if(TPDO->TPDOCommPar->inhibitTime && *timerNext_us > TPDO->inhibitTimer){
+                    *timerNext_us = (TPDO->inhibitTimer + 9) / 10;
+                }else if(TPDO->TPDOCommPar->eventTimer && *timerNext_us > TPDO->eventTimer){
+                    *timerNext_us = TPDO->eventTimer;
+                }
```

Comment:
Probably you mean * instead of + `(TPDO->inhibitTimer + 9) / 10`

I think, you can skip inhibit timer and use only:
```
                if(TPDO->TPDOCommPar->eventTimer && *timerNext_us > TPDO->eventTimer){
                    *timerNext_us = TPDO->eventTimer;
                }
```

Reply:
If there is an inhibit time set, and a COS or application requested send has been queued, do we not want it to go as soon as the inhibit time has expired? I think if we only did the eventTimer it would potentially not send as soon as it was allowed to when the application requests it.

For example, say the eventTimer is 10 seconds, but inhibit time is 1 second and it is configured for a COS send or every 10 seconds max. Then if it sends, the eventTimer is 10 seconds, but a COS occurs 0.25s after the initial send, then instead of waiting another 0.75s (the duration of the inhibit) and sending it would wait another 9.75s.

Or am I not understanding this correctly?

Reply:
Also, regarding 

> Probably you mean * instead of + `(TPDO->inhibitTimer + 9) / 10`

This was duplicating the ceiling function implemented in CO_Emergency.c here:

`326         diff = (emInhTime + 9) / 10; /* time difference in ms, always round up */`

So if that is actually wrong, then we should probably change it in that as well correct?

Reply:
I updated it to simply do the inhibit time plus 1 to put it past the inhibit window. Is that sufficient? I just realized that the CO_EM_process function apparently operates in 100us increments so I think I see where the confusion lies. Perhaps that's another reason to move to a common microsecond time base haha.